StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/CustomerSheet/CustomerSheet.swift

```diff
@@ -326,7 +326,25 @@ extension CustomerSheet {
         switch customerSheetDataSource.dataSource {
         case .customerSession(let customerSessionAdapter):
             let (elementsSession, customerSessionClientSecret) = try await customerSessionAdapter.elementsSessionWithCustomerSessionClientSecret()
-            let selectedPaymentOption = CustomerPaymentOption.defaultPaymentMethod(for: customerSessionClientSecret.customerId)
+
+            var selectedPaymentOption: CustomerPaymentOption?
+
+            // read from back end
+            if configuration.allowsSetAsDefaultPM,
+               let customer =  elementsSession.customer {
+                let defaultPaymentMethod = customer.paymentMethods.filter {
+                    $0.stripeId == customer.defaultPaymentMethod
+                }.first
+                if let defaultPaymentMethod = defaultPaymentMethod {
+                    selectedPaymentOption = CustomerPaymentOption.stripeId(defaultPaymentMethod.stripeId)
+                }
+                else {
+                    selectedPaymentOption = CustomerPaymentOption.defaultPaymentMethod(for: customerSessionClientSecret.customerId)
+                }
+            }
+            else {
+                selectedPaymentOption = CustomerPaymentOption.defaultPaymentMethod(for: customerSessionClientSecret.customerId)
+            }
```

Comment:
Having multiple code paths/branches that lead to the same code `CustomerPaymentOption.defaultPaymentMethod(for: customerSessionClientSecret.customerId)` is smelly. I think there's a better way to write this.

Any reason we can't do
```swift
if configuration.allowsSetAsDefaultPM,
   let customer = elementsSession.customer,
   let defaultPaymentMethodId = customer.defaultPaymentMethod,
   let defaultPaymentMethod = customer.paymentMethods.first(where: { $0.stripeId == defaultPaymentMethodId }) {
    selectedPaymentOption = .stripeId(defaultPaymentMethod.stripeId)
} else {
    selectedPaymentOption = .defaultPaymentMethod(for: customerSessionClientSecret.customerId)
}
```